File: quiz/deno/src/statemaschine/states/logged-in/logged-in-state.ts
import { Select } from "https://deno.land/x/cliffy@v1.0.0-rc.3/prompt/mod.ts";
import { User } from "../../../user.ts";
import { IState } from "../state.ts";
import { InitState } from "../init-state.ts";
import { ManageQuizzesState } from "../quiz/manage/manage-quizzes-state.ts";

export class LoggedInState implements IState {
  constructor(private readonly user: User) {}

  async run(): Promise<IState> {
    enum Action {
      ManageQuizzes = "manage-quizzes",
      Logout = "logout",
    }

    const action: string = await Select.prompt({
      "message": `Hello ${this.user.username}, what do you want to do?`,
      "options": [
        { name: "Manage Your Quizzes", value: Action.ManageQuizzes },
        { name: "Logout", value: Action.Logout },
      ],
    });

    switch (action) {
      case Action.ManageQuizzes:
        return new ManageQuizzesState(this.user);
      case Action.Logout:
        return new InitState();
    }

    return new InitState();
  }
}

File: quiz/go/statemaschine/states/logged-in-state.go
package states

import (
	"github.com/DoktorLenz/kp-wise23-24/quiz/go/user"
	"github.com/manifoldco/promptui"
)

type LoggedInState struct {
	user user.IUser
}

func (state *LoggedInState) Run() IState {
	prompt := promptui.Select{
		Label: "Hello " + state.user.GetUsername() + ", what do you want to do?",
		Items: []string{"Manage Your Quizzes", "Logout"},
	}

	_, result, _ := prompt.Run()

	switch result {
	case "Manage Your Quizzes":
		return &ManageQuizzesState{user: state.user}
	case "Logout":
		return &InitState{}
	}

	return &InitState{}
}
